src/mlpack/methods/ann/layer/subview.hpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file subview.hpp
+ * @author Haritha Nair
+ *
+ * Definition of the Subview class, which modifies the input as necessary.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SUBVIEW_HPP
+#define MLPACK_METHODS_ANN_LAYER_SUBVIEW_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/activation_functions/identity_function.hpp>
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * Implementation of the subview layer. The subview layer modifies the input to
+ * a submatrix of required size.
+ *
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class Subview
+{
+ public:
+  /**
+   * Create the Subview layer object using the specified range of input to accept.
+   *
+   * @param begin Start index.
+   * @param end End index.
```

Comment:
Looks like this works with 1D inputs, but it might be useful to call this `Subview1D`, and perhaps provide some other time a `Subview2D` layer, in case the input is matrix-shaped (not vector-shaped).

Reply:
Good point, it should be possible to support both in a single class, so I think we can still call this class ``Subview``.

Reply:
I think it does work for both vector and matrix shaped inputs. Maybe I understood something wrong, but I tried running it on matrices and it is being modified as required.

Reply:
Right, it works with both, but let's suppose I have a matrix of size 5x5 and what I want to do is take the middle 3x3 elements out of it.  If we use the Armadillo advanced constructors we don't have a way to do that, since the middle 3x3 won't be contiguous in memory.  In that situation we would have no choice but to extract the submatrix with `submat()`.  (I guess it might be possible to pass an arma::subview type to the next layer as input, but I am not sure if that would actually work in the framework we have.  I don't think that avoiding the subview copy in this situation is so important that it has to be dealt with right now though.)

Reply:
Right, then we can go on with `submat()`. I suppose. @zoq Unfortunately I deleted the branch I was working on, so just let me know if I should restore it and revert the merge to make these changes.

Reply:
No need to revert the changes, we can open a new PR to do the modification.